beauty-consulting-app/app/Pages/Messaging/ConversationsPage.js

```diff
@@ -1,20 +1,64 @@
-import React, { useContext } from "react";
-import { View, Text } from "react-native";
+import React, { useContext, useState, useEffect } from "react";
+import { View, Text, StyleSheet } from "react-native";
+import { useFocusEffect } from "@react-navigation/native";
 import SignupBackground from "../../assets/components/SignupBackground";
 import { UserContext } from "../../contexts/userContext";
+import api from "utils/axios";
+import handleHTTPError from "utils/errorHandling";
+import ConversationList from "../../assets/components/ConversationList";
 
 function ConversationsPage() {
 	const userContext = useContext(UserContext);
+	const [errorMessage, setErrorMessage] = useState("");
+	const [recentMessages, setRecentMessages] = useState();
+
+	const fetchMessages = async () => {
+		try {
+			const response = await api.get("/messages/recent", {
+				params: {
+					username: userContext.username,
+				},
+			});
+			setRecentMessages(response.data);
+		} catch (error) {
+			handleHTTPError(error, setErrorMessage);
+		}
+	};
+
+	useFocusEffect(
+		React.useCallback(() => {
+			fetchMessages();
+		}, [])
+	);
 
 	return (
 		<SignupBackground>
-			<View>
-				<Text>Conversations</Text>
-				<Text>
-					This is the conversations page for {userContext.username}
-				</Text>
+			<View style={styles.container}>
+				<Text style={styles.header}>Conversations</Text>
+				<ConversationList conversations={recentMessages} />
+				{errorMessage ? (
+					<Text style={styles.error}>{errorMessage}</Text>
```

Comment:
unless you need a specific kind of error (i havent looked extensively) we have an error message asset for this that implements this logic for you